src/load_order.ts

```diff
@@ -212,42 +219,41 @@ export const loadOrderUsageInstructionsForVortexGui =
 
 
 const deserializeLoadOrder = (vortexApi: VortexApi) =>
-  (loadOrderPathForCurrentProfile: string): Task<readonly LoadOrderEntry[]> => {
-
-    const decodedLoadOrder = pipe(
+  (loadOrderPathForCurrentProfile: string): TaskEither<Error, readonly LoadOrderEntry[]> =>
+    pipe(
       fileFromDiskTE({ relativePath: loadOrderPathForCurrentProfile, pathOnDisk: loadOrderPathForCurrentProfile }),
-      chainEitherK(({ content }) => decodeLoadOrder(content)),
-      matchTE(
-        (error) => {
-          vortexApi.log(`warn`, `${me}: Failed to deserialize load order, ignoring the file: ${error.message}`);
-          return [] as LoadOrderEntry[];
-        },
-        (loadOrder) => {
-          vortexApi.log(`debug`, `${me}: Current stored load order deserialized: ${S(loadOrder)}`);
-          return loadOrder.entriesInOrderWithEarlierWinning;
-        },
-      ),
+      swapTE,
+      mapTE((error) => {
+        vortexApi.log(`warn`, `${me}: Couldn't open load order file, proceeding with an empty list: ${error.message}`);
+        return [] as LoadOrderEntry[];
+      }),
+      orElseTE(({ content }) =>
+        pipe(
+          decodeLoadOrder(content),
+          fromEitherTE,
+          mapLeftTE((error) =>
+            new Error(`Couldn't decode load order file: ${error.message}`)),
+          mapTE((loadOrder) => {
+            vortexApi.log(`info`, `${me}: Successfully deserialized load order id: ${Date.parse(loadOrder.generatedAt)} (${loadOrder.generatedAt})`);
+            vortexApi.log(`debug`, `${me}: Current stored load order deserialized: ${S(loadOrder)}`);
+            return loadOrder.entriesInOrderWithEarlierWinning;
+          }),
+        )),
```

Comment:
This may warrant additional context: the first error is recoverable, the second isn't.

I failed to come up with a good way to handle that\*, so the `swap` turns the error into a default success and the `orElse` replaces the 'error' with a normal `[Task]Either`.

\* I guess cleanest would've been to write a blank LO but that's just no.

Reply:
I guess it is good enough for now. There is a lot of refactoring that can be done all over... _eventually_.